### PR TITLE
Introduce headerCase attribute in the CsvParserPipe

### LINKED ISSUE #1872: Invalid Console warnings on boolean attributes
**Describe the issue**
The consoles shows warnings that are most probably not correct: 
CsvParserPipe [csvToXml] does not have an attribute [fileContainsHeader] to set to value [true]

There is an attribute fileContainsHeader in the class CsvParserPipe.

And probably more console warnings on other boolean attributes.

**Reporter**
Richard van der Stel

**To Reproduce**

**Screenshots**
image

**Configuration**

            <pipe name="csvToXml"
                  className="nl.nn.adapterframework.pipes.CsvParserPipe"
                  fileContainsHeader="true"
                  fieldSeparator=";"
            >
                <forward name="exception"   path="fail:selectNumberOfQueryRecords"/>
                <forward name="success"     path="processBlocks"/>
            </pipe>

**Input**
```
Please provide an example of the input message. 
Alternatively, provide a Ladybug report.
```

**Environment**
FF! 7.7-20210527.152350: Ibis4Titan
Running on LPZWNL000022420 using Apache Tomcat/8.5.34
Java Version: Java(TM) SE Runtime Environment (1.8.0_60-b27)
Heap size: 1150M, total JVM memory: 1534M
Free disk space: 12GB, total disk space: 236GB
Up since: 2021-06-01 14:08:25 (20m)

**Additional Environment**
 - DBMS: [e.g. Oracle, MSSQL, MariaDB, MySql, PostgreSQL]
 - Browser: [e.g. Chrome, Safari, Edge, Firefox]

**Additional Context**
Add any other context about the problem here. (f.e. ladybug report / test adapter with larva test)

### LINKED ISSUE #1876: Option in CsvParserPipe to get upper/lower-case xml elements when using fileContainsHeader
**Describe the issue**
When the CsvParserPipe uses the fileContainsHeader option, the headernames are translated to XML elements.
However, when the headerline can contain upper and lowercase headers, the consecutive xslt parsers need to deal with these cases. It would be nice to have an option to get all the xml elements in upper-case, or lower-case, regardless of the case of the headernames. This way, the xslt always gets upper-case (or lowercase) without having to convert the case.

Example:
input: header1, HEADER2, HeAdEr3
 
ouput:
&lt;HEADER1/&gt;
&lt;HEADER2/&gt;
&lt;HEADER3/&gt;

Another new option:
when using fieldNames, to override the names in the headerline, the CsvParserPipe does not skip the headerline. It assumes that when using fieldNames there is no headerline. 
It would be nice to be able to have an option to skip the header line as well when using fieldNames to override the headernames.

**Reporter**
Richard van der Stel

**To Reproduce**

**Screenshots**

**Configuration**

**Input**

**Environment**
FF! 7.7-20210527.152350: Ibis4Titan
Running on LPZWNL000022420 using Apache Tomcat/8.5.34
Java Version: Java(TM) SE Runtime Environment (1.8.0_60-b27)
Heap size: 532M, total JVM memory: 1430M
Free disk space: 12GB, total disk space: 236GB
Up since: 2021-06-02 12:51:54 (43m)

**Additional Environment**
 - DBMS: [e.g. Oracle, MSSQL, MariaDB, MySql, PostgreSQL]
 - Browser: [e.g. Chrome, Safari, Edge, Firefox]

**Additional Context**
Add any other context about the problem here. (f.e. ladybug report / test adapter with larva test)

### PR DESCRIPTION
**Pull request by alisihab**
Closes #1876 
Closes #1872

### CODE REVIEW

File: core/src/main/java/nl/nn/adapterframework/pipes/CsvParserPipe.java [hunk 1]
```diff
@@ -107,4 +124,12 @@ public void setFieldSeparator(String fieldSeparator) {
 		this.fieldSeparator = fieldSeparator;
 	}
 
+	@IbisDoc({"4", "When set, character casing will be changed for the header. Possible values ['lowercase', 'uppercase']","null"})
```
Comment (gvanbrakel) [inline core/src/main/java/nl/nn/adapterframework/pipes/CsvParserPipe.java:127]: For the Frank Developer, the default value is not `null` but `not set`

File: core/src/test/java/nl/nn/adapterframework/configuration/digester/MapPropertyDescriptorsTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+			protected String buildDefaultBeanName(BeanDefinition definition) {
+				String beanClassName = definition.getBeanClassName();
+				Assert.state(beanClassName != null, "No bean class name set");
+				return beanClassName;
+			}
+		};
+		scanner.setBeanNameGenerator(beanNameGenerator);
+
+		int numberOfBeans = scanner.scan("nl.nn.adapterframework", "nl.nn.ibistesttool");
+		System.out.println("Found "+numberOfBeans+" beans registered!");
```
Comment (gvanbrakel) [inline core/src/test/java/nl/nn/adapterframework/configuration/digester/MapPropertyDescriptorsTest.java:44]: Please do not System.out.println(), but use log.debug()

File: core/src/test/java/nl/nn/adapterframework/pipes/CsvParserPipeTest.java [hunk 3]
```diff
@@ -70,6 +70,77 @@ public void testFieldSeparator() throws Exception {
...
+		configureAndStartPipe();
+		String csv ="A,b,c\n1,2,3\nx,\"y,y\"";
+		String expected="<csv><record><A>1</A><b>2</b><c>3</c></record><record><A>x</A><b>y,y</b></record></csv>";
+		
+		PipeRunResult prr = doPipe(csv);
+		assertXmlEquals(expected,prr.getResult().asString());
+	}
 
+	@Test
+	public void testUseFieldNameAsHeaderFileContainesHeaderFalse() throws Exception {
```
Comment (gvanbrakel) [inline core/src/test/java/nl/nn/adapterframework/pipes/CsvParserPipeTest.java:136]: Could you also add a test for FileContainsHeader=false without fieldNames specified?

**Review by gvanbrakel — CHANGES_REQUESTED** [review]

File: core/src/test/java/nl/nn/adapterframework/pipes/CsvParserPipeTest.java (hunk 3 above)
Comment (alisihab) [reply core/src/test/java/nl/nn/adapterframework/pipes/CsvParserPipeTest.java:136]: this test case exists, see line 34

**Review by nielsm5 — APPROVED** [review]

**Pull request merged**